### PR TITLE
Fix issue with decimal precision on manage stock modal and ledger exports [SCI-9646]

### PR DESCRIPTION
**Pull request by wandji20**
Jira ticket: [SCI-9646](https://scinote.atlassian.net/browse/SCI-9646)

### What was done
_Fix issue with decimal precision on the manage stock modal and ledger exportst_

#### ToDo:
_N/A_

### Note:
_N/A_

[SCI-9646]: https://scinote.atlassian.net/browse/SCI-9646?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/javascript/vue/repository_row/manage_stock_value_modal.vue [hunk 1]
```diff
@@ -175,14 +175,21 @@
         return unit ? unit[1] : ''
       },
       newAmount: function() {
+        const amount = isNaN(this.amount) ? 0 : this.amount;
+        let value
         switch (this.operation) {
           case 2:
-            if (this.amount) return parseFloat(this.stockValue.amount) + this.amount
+            console.log([ parseFloat(this.stockValue.amount), amount])
```
Comment (artoscinote) [inline app/javascript/vue/repository_row/manage_stock_value_modal.vue:182]: remove console.log

File: app/javascript/vue/repository_row/manage_stock_value_modal.vue [hunk 2]
```diff
@@ -175,14 +175,21 @@
...
           case 3:
-            if(this.amount) return parseFloat(this.stockValue.amount) - this.amount
+            value = parseFloat(this.stockValue.amount) - amount
+            break;
           default:
-            return this.amount
+            value = amount
+            break;
         }
+        return Math.round(value * 10**parseInt(this.stockValue.decimals)) / 10**parseInt(this.stockValue.decimals)
```
Comment (artoscinote) [inline app/javascript/vue/repository_row/manage_stock_value_modal.vue:192]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/EPSILON

Does this also need to be used here?

File: app/services/repository_stock_ledger_zip_export.rb [hunk 3]
```diff
@@ -88,5 +90,13 @@ def generate_record_data(record)
       row_data.insert(9, *breadcrumbs_data)
       row_data
     end
+
+    def format_amount(amount, precision)
+      return unless amount
+      return amount.to_s.split('.')[0] if precision.zero?
```
Comment (artoscinote) [inline app/services/repository_stock_ledger_zip_export.rb:96]: Not sure I understand why "amount" is a string in this case.

File: app/javascript/vue/shared/input.vue [hunk 4]
```diff
@@ -42,9 +41,11 @@
       autoFocus: { type: Boolean, default: false },
       error: { type: String, required: false }
     },
-    watch: {
-      value: function(newVal) {
-        this.inputValue = newVal;
+    computed: {
+      inputValue() {
```
Comment (artoscinote) [inline app/javascript/vue/shared/input.vue:45]: could `<input type="number"` be used here in any way?

**Review by artoscinote — CHANGES_REQUESTED** [review]

File: app/javascript/vue/repository_row/manage_stock_value_modal.vue (hunk 2 above)
Comment (wandji20) [reply app/javascript/vue/repository_row/manage_stock_value_modal.vue:192]: @artoscinote 199.773 - 177.9 = 21.87299999999999 on the brower.

File: app/javascript/vue/shared/input.vue (hunk 4 above)
Comment (wandji20) [reply app/javascript/vue/shared/input.vue:45]: I had difficulties handling decimal precisions if the input is of type number, so switched to the previous manage modal implementation using text type.

File: app/javascript/vue/repository_row/manage_stock_value_modal.vue [hunk 5]
```diff
@@ -215,7 +223,7 @@
           success: (result) => {
             this.repositoryRowName = result.repository_row_name
             this.stockValue = result.stock_value
-            this.amount = parseFloat(result.stock_value.amount)
+            this.amount = String(new Decimal(result.stock_value.amount || 0))
```
Comment (okriuchykhin) [inline app/javascript/vue/repository_row/manage_stock_value_modal.vue:226]: why do you cast it to string and then later compare to number?

File: app/services/repository_stock_ledger_zip_export.rb [hunk 6]
```diff
@@ -88,5 +90,12 @@ def generate_record_data(record)
       row_data.insert(9, *breadcrumbs_data)
       row_data
     end
+
+    def format_amount(amount, precision)
```
Comment (okriuchykhin) [inline app/services/repository_stock_ledger_zip_export.rb:94]: check formatted_value method in RepositoryStockValue

File: app/controllers/repositories_controller.rb [hunk 7]
```diff
@@ -369,7 +369,8 @@ def export_repository_stock_items
       RepositoryStockZipExportJob.perform_later(
         user_id: current_user.id,
         params: {
-          repository_row_ids: row_ids
+          repository_row_ids: row_ids,
+          decimal_precisions: @repository.repository_stock_column.metadata['decimals']
```
Comment (okriuchykhin) [inline app/controllers/repositories_controller.rb:373]: Don't pass it as params to the job, instead read the current value from inside the job

**Review by okriuchykhin — APPROVED** [review]

**Pull request merged**